Lib/reprlib.py

```diff
@@ -62,6 +63,23 @@ def repr1(self, x, level):
         else:
             return self.repr_instance(x, level)
 
+    def _join(self, pieces, level):
+        if self.indent is None:
+            return ', '.join(pieces)
+        if not pieces:
+            return ''
+        indent = self.indent
+        if not isinstance(indent, str):
+            if not isinstance(indent, int) or indent < 0:
+                raise ValueError(
+                    "Repr.indent must be None, str or non-negative int, not "
+                    + repr(indent)
+                )
+        if isinstance(indent, int):
+            indent *= ' '
+        sep = ',\n' + (self.maxlevel - level + 1) * indent
```

Comment:
I strongly recommend revising the error handling as follows. This raise the correct error class and message for the different cases (`TypeError` for passing an incompatible type, and `ValueError` for an inappropriate int value), use Pythonic EAFP, and `raise from` the actual error. And thanks to some cleanup and rationalization (using f-strings, avoiding an unnecessarily nested `if` statement, and a duplicate `isinstance` check) its only 1-2 lines longer.

```suggestion
        if isinstance(indent, int):
            if indent < 0:
                raise ValueError(
                    'Repr.indent cannot be negative int (was {indent!r})')
            indent *= ' '
        try:
            sep = ',\n' + (self.maxlevel - level + 1) * indent
        except TypeError as error:
            raise TypeError(
                f'Repr.indent must be of type None, str or int, not {type(indent)}'
            ) from error
```

Reply:
Yes, I like the distinction between `TypeError` and `ValueError` :+1: